resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -60,6 +61,22 @@ private[spark] class KubernetesClusterSchedulerBackend(
     removeExecutor(executorId, reason)
   }
 
+  /**
+    * Get an application ID associated with the job.
+    * This returns the string value of spark.app.id if set, otherwise
+    * the locally-generated ID from the superclass.
+    * @return The application ID
+    */
+  override def applicationId(): String = {
+    val appId = {
```

Comment:
Nit: looks like you can simplify this as `conf.getOption("spark.app.id").map(_.toString).getOrElse {super.applicationId}`.

Reply:
Cool, that looks better